src/components/Editor/Preview/index.js

```diff
@@ -47,10 +47,24 @@ class Preview extends PureComponent {
     self.onMouseDown = this.onMouseDown.bind(this);
   }
 
+  // To stop newly opened sources from inheriting the scroll-state
+  // of the currently open tab we force a scroll to top.
+  topScrollNewTab(dispSource) {
+    if (dispSource.get("loadedState") == "loaded") return;
+
+    const sourceLocation = dispSource.location;
+    if (!sourceLocation || !sourceLocation.line) {
+      sourceLocation = { line: 0, column: 0 };
```

Comment:
what does this do?

Reply:
It checks if the new document was called with a location. We only want to scroll to the top if there is no line set in the first place, otherwise it would overwrite an intended scroll.